code/modules/mob/living/simple_animal/hostile/ordeal/green/midnight.dm

```diff
@@ -204,9 +243,178 @@
+	//Now that we've stopped firing, let's prepare to deploy a squad of friendlies
+	DeployBotSquad()
+
+// This proc is used to fire a barrage of mini lasers, it is called several times during the helix's lasers being active. An alternative way to do this would be
+// placing it on Life() like Waxing does, but I only wanted it to happen while the Helix was firing.
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/FireLaserBarrage(list/valid_turfs)
+	for(var/turf/unfortunate_turf in valid_turfs)
+		if(prob(6))
+			addtimer(CALLBACK(src, PROC_REF(FireMiniLaser), unfortunate_turf), rand(1, 50))
+
+
+// The mini-lasers are code taken from Waxing of the Black Sun with some minor statistical tweaks.
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/FireMiniLaser(turf/target_turf)
+	new /obj/effect/temp_visual/helix_minilaser(target_turf)
+
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/FireMacroLaser(turf/target_turf)
+	new /obj/effect/temp_visual/helix_macrolaser(target_turf)
+
+//They are mainly a nuisance that causes people stacking in tight spots to move and shuffle and get eachother killed. Encourages not dumping BLACK armour for the ordeal
+/obj/effect/temp_visual/helix_minilaser
+	name = "helix of the end mini-laser"
+	icon = 'ModularTegustation/Teguicons/32x64.dmi'
+	icon_state = "pillar_strike"
+	duration = 15
+	color = "#7ac21f"
+
+/obj/effect/temp_visual/helix_minilaser/Initialize()
+	..()
+	addtimer(CALLBACK(src, PROC_REF(Blowup)), 10)
+
+/obj/effect/temp_visual/helix_minilaser/proc/Blowup()
+	playsound(src, 'sound/weapons/fixer/generic/rcorp4.ogg', 15, FALSE, 4)
+	for(var/mob/living/carbon/human/H in src.loc)
+		H.deal_damage(55, BLACK_DAMAGE)
+		to_chat(H, span_userdanger("You're hit by [src.name]!"))
+
+//This laser hits in a 3 tile radius (the epicenter and its adjacent tiles).
+/obj/effect/temp_visual/helix_macrolaser
+	name = "helix of the end macro-laser"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "warning"
+	duration = 21
+	color = "#7ac21f"
+	pixel_x = -32
+	base_pixel_x = -32
+	pixel_y = -32
+	base_pixel_y = -32
+
+/obj/effect/temp_visual/helix_macrolaser/Initialize()
+	..()
+	addtimer(CALLBACK(src, PROC_REF(Blowup)), 20)
+
+/obj/effect/temp_visual/helix_macrolaser/proc/Blowup()
+	playsound(src, 'sound/abnormalities/crying_children/sorrow_shot.ogg', 20, FALSE, 4)
+	for(var/mob/living/carbon/human/H in range(1, src))
+		H.deal_damage(50, BLACK_DAMAGE)
+		to_chat(H, span_userdanger("You're hit by [src.name]!"))
+
+
+//This proc is called to return a list with all the types that should be spawned by the next deployment. DeployBotSquad() trusts whatever list is given to it, so we
+//make sure we don't generate too many mobs in this proc (else midnight could spawn infinite mobs)
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/GenerateBotSquad()
+	var/list/squad = list()
+	var/remaining_spawn_budget = squad_size - active_minions
+	while(remaining_spawn_budget > 0)
```

Comment:
minor nitpick, `while()` loops should not really be used unless you know very well what you are doing.
Ya might wanna consider changing it for a `for()` loop instead, yet do feel free to just mark this as complete whenever you read it.

Reply:
It is because they can become infinite, that and its easier to read for() loops than while() loops.